waku/v2/node/rest/filter/handlers.nim

```diff
@@ -0,0 +1,151 @@
+when (NimMajor, NimMinor) < (1, 4):
+  {.push raises: [Defect].}
+else:
+  {.push raises: [].}
+
+import
+  std/sequtils,
+  stew/byteutils,
+  chronicles,
+  json_serialization,
+  json_serialization/std/options,
+  presto/route,
+  presto/common
+import
+  ../../../waku_core,
+  ../../../waku_filter,
+  ../../../waku_filter/client,
+  ../../message_cache,
+  ../../peer_manager,
+  ../../waku_node,
+  ../serdes,
+  ../responses,
+  ./types
+  
+export types
+
+logScope:
+  topics = "waku node rest filter_api"
+
+
+##### Topic cache
+
+const futTimeout* = 5.seconds # Max time to wait for futures
+
+
+#### Request handlers
+
+const ROUTE_FILTER_SUBSCRIPTIONSV1* = "/filter/v1/subscriptions"
+
+const filterMessageCacheDefaultCapacity* = 30
+
+type
+  MessageCache* = message_cache.MessageCache[ContentTopic]
+
+
+func decodeRequestBody[T](contentBody: Option[ContentBody]) : Result[T, RestApiResponse] =
+  # Check the request body
+  if contentBody.isNone():
+    return err(RestApiResponse.badRequest())
+
+  let reqBodyContentType = MediaType.init($contentBody.get().contentType)
+  if reqBodyContentType != MIMETYPE_JSON:
+    return err(RestApiResponse.badRequest())
+
+  let reqBodyData = contentBody.get().data
+
+  let reqResult = decodeFromJsonBytes(T, reqBodyData)
+  if reqResult.isErr():
+    return err(RestApiResponse.badRequest())
+
+  return ok(reqResult.get())
+
+proc installFilterPostSubscriptionsV1Handler*(router: var RestRouter, node: WakuNode, cache: MessageCache) =
+
+  router.api(MethodPost, ROUTE_FILTER_SUBSCRIPTIONSV1) do (contentBody: Option[ContentBody]) -> RestApiResponse:
+    # ## Subscribes a node to a list of contentTopics of a pubsubTopic
+    # debug "post_waku_v2_filter_v1_subscriptions"
+
+    let decodedBody = decodeRequestBody[FilterSubscriptionsRequest](contentBody)
+
+    if decodedBody.isErr():
+      error "Failed to decode body", error=decodedBody.error()
+      return decodedBody.error()
```

Comment:
Shall we have better feedback in case of error?

```suggestion
      let msg = "Failed to decode body: " & decodedBody.error()
      error msg
      return RestApiResponse.badRequest(msg)
```